### PR TITLE
Typing: Evaluator

### PR DESCRIPTION
**Pull request by asi1024**
Depends on https://github.com/pfnet/pytorch-pfn-extras/pull/274.

### CODE REVIEW

**Comment by emcastillo** [comment]
rebase please :)

**Comment by kmaehashi** [comment]
I've resolved conflicts.
/test

File: pytorch_pfn_extras/training/_evaluator.py [hunk 1]
```diff
@@ -86,9 +107,9 @@ def run(
                 The number of iterations per one evaluation epoch.
         """
         # Note: setup_manager is done by the Trainer.
-        self._idxs = queue.Queue()
-        self._inputs = queue.Queue()
-        self._observed = queue.Queue()
+        self._idxs = queue.Queue[int]()
+        self._inputs = queue.Queue[DictBatch]()
+        self._observed = queue.Queue['Observation']()
```
Comment (kmaehashi) [inline pytorch_pfn_extras/training/_evaluator.py:112]: E       TypeError: 'type' object is not subscriptable

**Comment by emcastillo** [comment]
```


00:03:53.274372 STDOUT 2343] | >       self._update = self._pbar.__enter__() |  
-- | -- | --
00:03:53.274375 STDOUT 2343] | E       AttributeError: 'generator' object has no attribute '__enter__'


```

File: pytorch_pfn_extras/training/_evaluator.py [hunk 2]
```diff
@@ -1,20 +1,33 @@
...
+    from pytorch_pfn_extras.runtime._runtime import Batch
+    from pytorch_pfn_extras.training.metrics import MetricType
+    from pytorch_pfn_extras.reporting import Observation
+
+
+def _progress_bar(
+        name: str,
+        required: bool,
+        size: int,
+) -> Generator[Callable[[int], None], None, None]:
```
Comment (emcastillo) [inline pytorch_pfn_extras/training/_evaluator.py:25]: this return type seems to be causing a problem :(

**Comment by asi1024** [comment]
PTAL.

**Review by emcastillo — APPROVED** [review]
LGTM

**Pull request merged**